lib/rules/no-unreachable.js

```diff
@@ -183,10 +189,29 @@ module.exports = {
             // Manages the current code path.
             onCodePathStart(codePath) {
                 currentCodePath = codePath;
```

Comment:
I think we can remove `currentCodePath` variable as it's basically unused after this change.